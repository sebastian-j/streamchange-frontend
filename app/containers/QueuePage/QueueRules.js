import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import {
  makeSelectCapacity,
  makeSelectQueueCommand,
  makeSelectTTI,
  makeSelectTTK,
  makeSelectWidgetCode,
} from './selectors';
import {
  changeCapacity,
  changeQueueCommand,
  changeTTI,
  changeTTK,
  changeWidgetCode,
} from './actions';
import Panel from '../../components/Panel';
import PanelTitle from '../../components/Panel/PanelTitle';
import StyledTextField from '../../components/StyledTextField';
import AdFrame from '../../components/AdFrame';
import QueueWidgetDialog from './QueueWidgetDialog';

export const QueueRules = props => {
  const handleInputValueChange = event => {
    const { target } = event;
    const { value } = target;
    const { name } = target;

    if (value.length < 140 && name === 'command')
      props.changeQueueCommand(value);
    if (name === 'capacity') props.changeCapacity(value);
    if (name === 'timeToIdle') props.changeTTI(value);
    if (name === 'timeToKick') props.changeTTK(value);
    if (name === 'widgetCode') {
      props.changeWidgetCode(value);
      return;
    }
    localStorage.setItem(`queue-${name}`, value);
  };

  return (
    <Panel>
      <PanelTitle>
        <FormattedMessage {...messages.rulesPanelTitle} />
      </PanelTitle>
      <FormattedMessage {...messages.commandTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="command"
            onChange={handleInputValueChange}
            label={label}
            type="text"
            value={props.command}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.capacityTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="capacity"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={props.capacity}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.timeToIdleTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="timeToIdle"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={Number(props.timeToIdle)}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.timeToKickTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="timeToKick"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={Number(props.timeToKick)}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.widgetCodeTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="widgetCode"
            onChange={handleInputValueChange}
            label={label}
            type="password"
            value={props.widgetCode}
            fullWidth
          />
        )}
      </FormattedMessage>
      <QueueWidgetDialog />
      <AdFrame />
    </Panel>
  );
};

QueueRules.propTypes = {
  changeCapacity: PropTypes.func.isRequired,
  changeQueueCommand: PropTypes.func.isRequired,
  changeTTI: PropTypes.func.isRequired,
  changeTTK: PropTypes.func.isRequired,
  changeWidgetCode: PropTypes.func.isRequired,
  capacity: PropTypes.number.isRequired,
  command: PropTypes.string,
  timeToIdle: PropTypes.number,
  timeToKick: PropTypes.number.isRequired,
  widgetCode: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  capacity: makeSelectCapacity(),
  command: makeSelectQueueCommand(),
  timeToIdle: makeSelectTTI(),
  timeToKick: makeSelectTTK(),
  widgetCode: makeSelectWidgetCode(),
});

export function mapDispatchToProps(dispatch) {
  return {
    changeCapacity: cap => dispatch(changeCapacity(cap)),
    changeQueueCommand: command => dispatch(changeQueueCommand(command)),
    changeTTI: s => dispatch(changeTTI(s)),
    changeTTK: s => dispatch(changeTTK(s)),
    changeWidgetCode: code => dispatch(changeWidgetCode(code)),
    dispatch,
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(QueueRules);
