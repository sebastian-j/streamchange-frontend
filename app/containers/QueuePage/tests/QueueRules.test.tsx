import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import QueueRules from '../QueueRules';
import configureStore from '../../../configureStore';

describe('<QueueRules />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <QueueRules />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
