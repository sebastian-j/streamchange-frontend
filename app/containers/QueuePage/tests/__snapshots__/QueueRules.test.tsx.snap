// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueueRules /> should render and match the snapshot 1`] = `
<div
  class="sc-bdvvtL edLmX"
>
  <h2
    class="sc-gsDKAQ gzLYbA"
  >
    Queue rules
  </h2>
  <div
    aria-label="keyword"
    class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root sc-hKwDye fWArNR css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    data-mui-internal-clone-element="true"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="mui-1"
      id="mui-1-label"
    >
      Command for joining the queue
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-1"
        name="command"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    aria-label="keyword"
    class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root sc-hKwDye fWArNR css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    data-mui-internal-clone-element="true"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-3"
      id="mui-3-label"
    >
      Queue capacity
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-3"
        name="capacity"
        type="number"
        value="10"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root sc-hKwDye fWArNR css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-5"
      id="mui-5-label"
    >
      Time without any message to mark as inactive (sec)
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-5"
        name="timeToIdle"
        type="number"
        value="600"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root sc-hKwDye fWArNR css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-6"
      id="mui-6-label"
    >
      Time without any message to kick (sec)
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-6"
        name="timeToKick"
        type="number"
        value="900"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root sc-hKwDye fWArNR css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-7"
      id="mui-7-label"
    >
      Code for OBS widget with queue
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-7"
        name="widgetCode"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    style="display: inline-block;"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root sc-fotOHu jklEfd css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Queue widget for OBS
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    <div
      class="sc-hBUSln iXsoMr"
    >
      <span>
        This is a place for your advertisement. Prices starts from 10 PLN per day. Your ad will reach thousands of viewers.
      </span>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Show the details
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
